[CI] CoreWithSecurityClientYamlTestSuiteIT test {yaml=indices.simulate_index_template/10_basic/Simulate index template specifying a new template} failure

The following three tests failed on master intake CI and they are **reproducible**.
* test {yaml=indices.simulate_index_template/10_basic/Simulate index matches overlapping V1 and V2 templates}
* test {yaml=indices.simulate_index_template/10_basic/Simulate index template without new template in the body}  
* test {yaml=indices.simulate_index_template/10_basic/Simulate index template specifying a new template}

Build scan: https://gradle-enterprise.elastic.co/s/6tywp5h4mut6c

Can be reproduced with: 
> ./gradlew ':x-pack:qa:core-rest-tests-with-security:integTestRunner' --tests "org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT" -Dtests.method="test {yaml=indices.simulate_index_template/10_basic/Simulate index template specifying a new template}" -Dtests.seed=4221B011525C40AC -Dtests.security.manager=true -Dtests.locale=de-AT -Dtests.timezone=Africa/Lubumbashi -Dcompiler.java=14 -Druntime.java=11 -Dtests.rest.blacklist='index/10_with_id/Index with
 ID,indices.get_alias/10_basic/Get alias against closed indices'

The failure is specific to the seed. It passes with another seed. 

Sample error message:
```
java.lang.AssertionError: 
Failure at [indices.simulate_index_template/10_basic:40]: got unexpected warning header [
	299 Elasticsearch-8.0.0-SNAPSHOT-03cd802f460e8a67fa69bdb40ce3e3edd62e6894 "index template [existing_test] has index patterns [te*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [existing_test] will take precedence during new index creation"
]
```

--- a/rest-api-spec/src/main/resources/rest-api-spec/test/indices.simulate_index_template/10_basic.yml
+++ b/rest-api-spec/src/main/resources/rest-api-spec/test/indices.simulate_index_template/10_basic.yml
@@ -3,11 +3,9 @@
   - skip:
       version: " - 7.99.99"
       reason: "simulate index template API has not been backported"
-      features: allowed_warnings
+      features: ["default_shards"]
 
   - do:
-      allowed_warnings:
-        - "index template [test] has index patterns [te*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: test
         body:
@@ -35,11 +33,9 @@
   - skip:
       version: " - 7.99.99"
       reason: "simulate index template API has not been backported"
-      features: allowed_warnings
+      features: ["default_shards"]
 
   - do:
-      allowed_warnings:
-        - "index template [test] has index patterns [te*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [test] will take precedence during new index creation"
       indices.put_index_template:
         name: existing_test
         body:
@@ -122,7 +118,7 @@
   - skip:
       version: " - 7.99.99"
       reason: "simulate index template API has not been backported"
-      features: allowed_warnings
+      features: ["allowed_warnings", "default_shards"]
 
   - do:
       indices.put_template:

